File: Easy/14_LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        common_prefix = strs[0]
        for i in range(1,len(strs)):
            temp = ""
            if len(common_prefix) == 0:
                break
            for j in range(len(strs[i])):
                if j<len(common_prefix) and common_prefix[j] == strs[i][j]:
                    temp+=common_prefix[j]
                else:
                    break
            common_prefix = temp
        return common_prefix

File: Easy/14_LongestCommonPrefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    return "" if strs.empty?

    prefix = strs[0]

    strs.each do |str|
        while str.index(prefix) != 0
            prefix = prefix[0...-1]
            return "" if prefix.empty?
        end
    end

    prefix
end
